Tests/timestamp_replacer.py

```diff
@@ -236,95 +248,100 @@ def modify_json_body(self, req: HTTPRequest, json_body: dict) -> None:
+        logging.info('query_data: {}'.format(query_data))
         for key, val in query_data:
             # don't bother trying to interpret an argument less than 4 characters as some type of timestamp
             if len(val) > 4:
-                if parse(val):
+                if self.safely_parse(val):
                     self.query_keys.add(key)
 
     def handle_multipart_form(self, req: HTTPRequest) -> None:
-        '''Used when detecting what keys in a multipart form to replace with constants.
+        """Used when detecting what keys in a multipart form to replace with constants.
 
         Args:
             req (HTTPRequest): The request to inspect
-        '''
+        """
         if req.multipart_form:
             for key, val in req.multipart_form.items(multi=True):
                 # don't bother trying to interpret an argument less than 4 characters as some type of timestamp
                 if len(val) > 4:
-                    if parse(val):
+                    if self.safely_parse(val):
                         self.form_keys.add(key)
 
     def handle_urlencoded_form(self, req: HTTPRequest) -> None:
-        '''Used when detecting what keys in an url encoded parameters to replace with constants.
+        """Used when detecting what keys in an url encoded parameters to replace with constants.
 
         Args:
             req (HTTPRequest): The request to inspect
-        '''
+        """
         if req.urlencoded_form:
             for key, val in req.urlencoded_form.items(multi=True):
                 # don't bother trying to interpret an argument less than 4 characters as some type of timestamp
                 if len(val) > 4:
-                    if parse(val):
+                    if self.safely_parse(val):
                         self.form_keys.add(key)
 
     def handle_json_body(self, req: HTTPRequest) -> None:
-        '''Used when detecting what keys in a request's json body to replace with constants.
+        """Used when detecting what keys in a request's json body to replace with constants.
 
         Args:
             req (HTTPRequest): The request to inspect
-        '''
+        """
         if req.method == 'POST':
             raw_content = req.raw_content
             if raw_content is not None:
                 content = req.raw_content.decode()
             else:
                 content = ''
+            logging.info(f'handling json body: content={content}')
             json_data = content.startswith('{')
             if json_data:
-                content = json.loads(content, object_pairs_hook=OrderedDict)
-                json_keys = self.determine_problematic_keys(content)
-                self.json_keys.update(json_keys)
+                try:
+                    content = OrderedDict(literal_eval(content))
+                    json_keys = self.determine_problematic_keys(content)
+                    self.json_keys.update(json_keys)
+                except Exception:
```

Comment:
Let's handle the particular exception that `literal_eval` raises when it can't parse the argument it receives. Otherwise, let's log the exception itself and the value of `content` instead of assuming it was an exception from `ast.literal_eval`. This comment is also applicable to the `clean_json_body` function.

Reply:
When you do logging.exception it will also log the exception trace back and message.
All I'm doing is adding the content it was attempted on in the beggining so we can debug.
I think we would want to see that exception traceback if it was for literal_eval or for determine_problematic_keys

Reply:
I agree that we should output the value of content - my issue is that you are catching all exceptions, and then your exception message could be misleading because it assumes that it is an exception raised by `literal_eval`, regardless of what actually raised the exception. My initial suggestion wasn't that you shouldn't log `content` when handling the exception raised by `literal_eval`. We should always log it. I think you should either implement my initial suggestion or you should change the message you log so that it doesn't assume the exception stems from `literal_eval`.

Reply:
@avidan-H Changed the message.

Although just to make sure we understand it - you will see the trace, so even if the message was misleading - the trace will show you excactly what kind of exception it was and why it was raised.